src/backend/common/UsageTracker.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 The NXT Authors
```

Comment:
I'd be interested to see how tracking inside the command buffer and outside can be factored, it seems to me the additional tracking inside is much simpler (no freezing involved).
Also I'm not a huge fan of using maps to store temporary information. What do you think of having a scratch buffer inside resources that can be used to store information temporarily for validation? This way we would avoid allocations, hashing pointers and maybe reduce the number of cache lines we access.